### PR TITLE
Fixing distribution of otioClip attribute to plates

### LINKED ISSUE #30: AY-7142_broken vertical alignment resource distribution
### Please describe the issue you have encountered?

Client was reporting that plate resources were not correctly distributed within vertically aligned workflow. Hero plate media source was cloned into all other child plates.

### How to replicate the issue?

_No response_

### Additional context:
link to discussion on [Discord](None)
_(might be a private channel)_

_This issue was automatically created from Clickup ticket [AY-7142](https://app.clickup.com/t/86c10u5kv)_

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Vertical aligned shot resources are distributed correctly now. 

## Testing notes:
1. Create timeline in Hiero with vertically aligned shots in multiple tracks. Make sure media resources for each clip is different.
2. Rename tracks to hold some unique names
3. select shots which are vertically aligned
4. go to AYON menu and Create
5. enamble vertical sync
6. Select hero track 
7. Hit Create and then Publish.
8. notice the integrated files and make sure plate version is having correct representation files. 

Resolves #30

### CODE REVIEW

**Review by BigRoy — APPROVED** [review]
Confirmed by client that it works.

Does this issue maybe also influence other parts of the code, like collect audio?

**Review by robin-ynput — APPROVED** [review]

**Comment by robin-ynput** [comment]
Sorry for this I realize this was my mistake. I have reported the fix for Resolve here: https://github.com/ynput/ayon-resolve/pull/41 and will ensure it works properly in Flame as well.

File: client/ayon_hiero/plugins/publish/collect_shots.py [hunk 1]
```diff
@@ -22,7 +22,6 @@ class CollectShot(pyblish.api.InstancePlugin):
         "handleStart",
         "handleEnd",
         "item",
-        "otioClip",
```
Comment (robin-ynput) [inline client/ayon_hiero/plugins/publish/collect_shots.py:25]: this'd need to be maintained for the `Collect Audio` plugin.

**Review by robin-ynput — CHANGES_REQUESTED** [review]

File: client/ayon_hiero/plugins/publish/collect_plates.py [hunk 2]
```diff
@@ -16,6 +17,17 @@ def process(self, instance):
         """
         instance.data["families"].append("clip")
 
+        # Adjust instance data from parent otio timeline.
+        otio_timeline = instance.context.data["otioTimeline"]
+        otio_clip, _ = utils.get_marker_from_clip_index(
+            otio_timeline, instance.data["clip_index"]
+        )
+        if not otio_clip:
+            raise RuntimeError(
```
Comment (iLLiCiTiT) [inline client/ayon_hiero/plugins/publish/collect_plates.py:26]: Requires import `from ayon_core.pipeline import PublishError`.
```suggestion
            raise PublishError(
```

**Review by robin-ynput — APPROVED** [review]

**Pull request merged**